Sources/ApodiniDeploy/RemoteHandlerInvocationManager.swift

```diff
@@ -0,0 +1,308 @@
+            return eventLoop.makeFailedFuture(ApodiniDeployError(message: "unable to get \(ApodiniDeployInterfaceExporter.self) object"))
+        }
+        
+        guard let targetEndpoint: Endpoint<H> = RHIIE.getEndpoint(withIdentifier: handlerId, ofType: H.self) else {
+            return eventLoop.makeFailedFuture(ApodiniDeployError(message: "Unable to find target endpoint. (handlerId: '\(handlerId)', handlerType: '\(H.self)')"))
+        }
+        
+        switch dispatchStrategy(forInvocationOf: targetEndpoint, RHIIE: RHIIE) {
+        case .locally:
+            return targetEndpoint._invoke(withCollectedParameters: collectedInputParams, RHIIE: RHIIE, on: eventLoop)
+        case .remotely(let targetNode):
+            guard let runtime = RHIIE.deploymentProviderRuntime else {
+                return eventLoop.makeFailedFuture(ApodiniDeployError(message: "Unable to find runtime"))
+            }
+            // The targetEndpoint is on a different node, dispatch the invocation there
+            
+            // TODO if theres a guarantee that an EndpointParameter's id is the same as the id of the @Parameter it was generated from,
+            // we can combine these two sets into one
+            var alreadyProcessedParamKeyPaths: Set<AnyKeyPath> = []
+            var alreadyProcessedEndpointParamIds: Set<UUID> = []
+
+            let invocationParams: [HandlerInvocation<H>.Parameter] = collectedInputParams.map { collectedParam in
+                // The @Parameter property wrapper declaration in the handler
+                let handlerParamId = (targetEndpoint.handler[keyPath: collectedParam.handlerKeyPath] as! AnyParameterID).value
+                let endpointParam = targetEndpoint.findParameter(for: handlerParamId)!
+                if !alreadyProcessedParamKeyPaths.insert(collectedParam.handlerKeyPath).inserted {
+                    print("Parameter '\(endpointParam.name)' specified multiple times in remote handler invocation")
+                }
+                if !alreadyProcessedEndpointParamIds.insert(endpointParam.id).inserted {
+                    print("Endpoint parameter with id '\(endpointParam.id)' matched multiple times in remote handler invocation")
+                }
+                return HandlerInvocation<H>.Parameter(
+                    stableIdentity: endpointParam.lk_stableIdentity,
+                    name: endpointParam.name,
+                    value: collectedParam.value as! HandlerInvocation<H>.Parameter.Value
+                )
+            }
+            let missingParamNames: [String] = targetEndpoint.parameters
+                .filter { !$0.hasDefaultValue && !alreadyProcessedEndpointParamIds.contains($0.id) }
+                .map(\.name)
+            guard missingParamNames.isEmpty else {
+                fatalError("Missing parameters in remote handler invocation: \(missingParamNames)")
```

Comment:
In this case we technically wouldn't throw, but return a failed future.
I'm currently a bit careful about returning failed futures, because that's also how errors thrown in handlers are returned to the invoking handler, and otherwise there'd be no distinction between "the remote handler invocation API returned a failed future because the called handler threw an error" and "the remote handler invocation API returned a failed future because something else internally went wrong".
For example, none of the remote-handler-invocation functions (most of them are called something alone the lines of `invoke`) currently throw, because otherwise the user would have to handle both the remoteHandlerAPI throwing, as well as it returning a failed future

Reply:
> otherwise there'd be no distinction between "the remote handler invocation API returned a failed future because the called handler threw an error" and "the remote handler invocation API returned a failed future because something else internally went wrong".

I understand, this makes sense. From a perspective of a developer using the invocation API the only really way to check this would be to inspect the error and see if the type is an `ApodiniError `(error thrown by a `Handler` if it follows the Apodini conventions) or any other error that would point to an error that most likely is based on an implementation mistake. I think this is a general thing that he exporters should consider, e.g. only forward ApodiniErrorrs to an API consumer and log all other errors internally and only return e.g. an 500 internal server error code for REST to the consumer. It would be good to note this in an GitHub issue so we take a closer look at this in the future.